external-api/src/helpers/cron.ts

```diff
@@ -0,0 +1,75 @@
+import dotenv from 'dotenv';
+import { CronJob } from 'cron';
+import { Estore } from '../interfaces';
+import { Category } from './types/estore';
+import { eStoreSiteCreationCron } from '../cron/eStoreSiteCreationJob.cron';
+
+dotenv.config();
+
+const { ESTORE_CRON_MANAGER_SCHEDULE, TZ } = process.env;
+const jobs = new Map();
+
+/**
+ * Executes a cron job based on the provided eStore data and category.
+ * @param eStoreData - The eStore data.
+ * @returns A promise that resolves when the cron job is completed.
+ */
+const onTick = async (eStoreData: Estore) => {
+  await eStoreSiteCreationCron(eStoreData);
+};
+
+/**
+ * Handles the completion of the eStore CRON for a specific deal category.
+ * @param eStoreData - The eStore data object.
+ * @param category - The category of the deal.
+ * @returns A promise that resolves when the completion is handled.
+ */
+const onComplete = (eStoreData: Estore, category: Category) => {
+  console.info('✅ eStore %s CRON has been completed successfully for deal %s', category, eStoreData.dealId);
+};
+
+/**
+ * Creates and starts a CRON job for eStore data.
+ * @param eStoreData - The eStore data.
+ * @param category - The category of the eStore data.
+ * @returns A promise that resolves when the CRON job is started.
+ */
+export const cron = (eStoreData: Estore, category: Category, kill: boolean = false): Promise<boolean> => {
```

Comment:
What part of this is asynchronous? Or is it just that the place we call this we have to pass in something that returns a Promise<Boolean>?

Reply:
Good catch, have removed `Promise` return.